### PR TITLE
docs[esl-related-target]: update tests, docs and demo pages

### LINKED ISSUE #1292: [📖 esl-related-target]: updates for module documentation and demo pages

### PR DESCRIPTION
**Pull request by NastaLeo**
Closes #1292

### CODE REVIEW

File: src/modules/esl-related-target/test/related-target.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    test('SHOW_REQUEST on the current mixin element leads to show target element', () => {
+      $el.hide();
+      jest.advanceTimersByTime(1);
+      expect($el.open).toBe(false);
+      expect($relatedEl.open).toBe(false);
+      $el.dispatchEvent(new CustomEvent($el.SHOW_REQUEST_EVENT));
+      jest.advanceTimersByTime(1);
+      expect($el.open).toBe(true);
+      expect($relatedEl.open).toBe(true);
+    });
```
Comment (dshovchko) [inline src/modules/esl-related-target/test/related-target.test.ts:29]: This test checks two expectations so I think it should be split into 2 different tests. 
By default Jest runs all the tests serially in the order they were encountered in the collection phase, waiting for each to finish and be tidied up before moving on. So just split this test into 2 tests.

File: src/modules/esl-related-target/test/related-target.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    test('HIDE_REQUEST on the current mixin element leads to show target element', () => {
+      $el.show();
+      jest.advanceTimersByTime(1);
+      expect($el.open).toBe(true);
+      expect($relatedEl.open).toBe(true);
+      $el.dispatchEvent(new CustomEvent($el.HIDE_REQUEST_EVENT));
+      jest.advanceTimersByTime(1);
+      expect($el.open).toBe(false);
+      expect($relatedEl.open).toBe(false);
+    });
```
Comment (dshovchko) [inline src/modules/esl-related-target/test/related-target.test.ts:42]: Same here

File: src/modules/esl-related-target/test/related-target.test.ts (hunk 1 above)
Comment (NastaLeo) [reply src/modules/esl-related-target/test/related-target.test.ts:29]: I'll split it)

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**